### DISCUSSION TITLE
Overload for useQuery when passing options for query with no params

### DISCUSSION THREAD

**Discussion by agawley** [trpc/trpc#3247] [Ideas]
This may be a dumb idea or not really feasible but I thought I'd raise it.

Currently when using useQuery on an endpoint that has no input, you have to pass undefined as the first parameter if you want to also pass opts:

`trpc.proxy.settings.get.useQuery(undefined, { staleTime: 10 })
`

The undefined just bothers me and I was wondering if it made sense to have an overload for useQuery that allows the following in the case of no input:

`trpc.proxy.settings.get.useQuery({ staleTime: 10 })
`

React-query doesn't have this issue since their first argument is a combined path & input array and is required. It's not a big deal but it does bother me a little as seeing undefined in my code normally means I am hacking something... :-)

**Comment by KATT** [comment]
No plans on changing this at the moment.

**Comment by kaminskypavel** [comment]
TIL that you can use `void 0` in javascript 🤷

```
  const { data } = trpc.settings.list.useQuery(void 0, {
    refetchOnWindowFocus: false,
  });
```

**Reply by marcospgp** [reply]
that is just undefined. it's something compilers use(d) because undefined could be overridden I think, also when minifying code cause it's shorter

**Comment by tahavirdiwala** [comment]
work for me

**Comment by marcospgp** [comment]
the undefined bothers me too!!! help us @KATT
